File: 028. Implement strStr()/C++/Implement strStr().cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        for (int i = 0; i < (int) haystack.size() - (int) needle.size() + 1; i++) {
            int j = 0;
            while (j < needle.size()) {
                if (haystack[i + j] != needle[j]) {
                    break;
                }
                j++;
            }
            if (j == needle.size()) {
                return i;
            }
        }
        return -1;
    }
};

File: 028. Implement strStr()/Python/Implement strStr().py
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        for i in range(0, len(haystack) - len(needle) + 1):
            j = 0
            while j < len(needle):
                if haystack[i + j] != needle[j]:
                    break
                j += 1
            if j == len(needle):
                return i
        return -1
